task/acs-determine-image-tag/0.1/README.md

```diff
@@ -0,0 +1,23 @@
+# acs-determine-image-tag task
```

Comment:
Obviously, it would be good to find out a convention suggested by the Konflux team about the way to store tenant-specific trusted tasks so that they don't show up in the list of shared tasks and don't confuse other tenants. Here, additionally, the task starts with `acs-` just as well as the existing `acs-deploy-check`, `acs-image-check` and `acs-image-scan`. The thing is that `acs-determine-image-tag` in this PR is _tenant-specific_ one which we need in order to have _our_ ACS builds compliant, the other three are _shared_ tasks that any tenant can use to leverage ACS features in their pipeline. The fact that they all will sit in the same directory with the same `acs-` prefix could be confusing.

Reply:
Ideally, we don't want these in build-definitions at all. It should now be possible to put custom, "untrusted" tasks in your pipeline and pass EC checks as long as those tasks do not modify the source code on the way to the build task.

That requires using a Trusted Artifacts-based pipeline. Example pipeline here: https://github.com/konflux-ci/olm-operator-konflux-sample/blob/main/.tekton/single-arch-build-pipeline.yaml

@konflux-ci/mota could probably provide more info.

Reply:
This task does not modify the source code. It gets the source code on the input and outputs an image tag that we put on the resulting images.

We already use oci-ta tasks and not using workspaces to pass the source code around.

We, however, have some other tasks that download blobs that are included in the source code and then included in the resulting containers.

What would be the plan? Should we skip this task but open PRs for the others that download blobs?

Reply:
> Should we skip this task but open PRs for the others that download blobs?

If it's generically re-usable, that may be reasonable. But the blobs downloaded by such a task would likely bypass source containers and SBOMs, wouldn't they?

The prefetch task should eventually gain the ability to download arbitrary blobs (while also allowing one to reference the source code for those blobs): https://issues.redhat.com/browse/KONFLUX-2390

Reply:
Until then, you might as well make your build non-hermetic

Reply:
> the blobs downloaded by such a task would likely bypass source containers and SBOMs, wouldn't they?

During ACS CPaaS onboarding, we had a session or two about these blobs. I could find [this doc](https://docs.google.com/document/d/1u9E9r-qeZyUZjfveMd6-yBalQu_5lNkAkPnflbs3JLQ/edit) but I think there should be more records and I can dig it out if needed.
Basically, we concluded that since these blobs carry data and no code, we can not include them in source containers.
In present Konflux time, the arrangements might be different and it seems Curlito could fit us.

> https://issues.redhat.com/browse/KONFLUX-2390

Thanks, I subscribed to it and linked to ACS Enablement ticket (KONFLUX-258).